tests/Kron.py

```diff
@@ -37,14 +39,111 @@
 
 #-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
 
-class KronPhotometryTestCase(unittest.TestCase):
+def makeGalaxy(width, height, flux, a, b, theta, dx=0.0, dy=0.0, xy0=None, xcen=None, ycen=None):
+    """Make a fake galaxy image"""
+    gal = afwImage.ImageF(width, height)
+    if xcen is None:
+        xcen = 0.5*width + dx
+    if ycen is None:
+        ycen = 0.5*height + dy
+    I0 = flux/(2*math.pi*a*b)
+
+    if xy0 is not None:
+        gal.setXY0(xy0)
+
+    c, s = math.cos(math.radians(theta)), math.sin(math.radians(theta))
+    I, Iuu, Ivv = 0.0, 0.0, 0.0
+    for y in range(height):
+        for x in range(width):
+            dx, dy = x + gal.getX0() - xcen, y + gal.getY0() - ycen
+            if math.hypot(dx, dy) < 10.5:
+                nsample = float(5)
+                subZ = np.linspace(-0.5*(1 - 1/nsample), 0.5*(1 - 1/nsample), nsample)
+            else:
+                nsample = 1
+                subZ = [0.0]
+
+            val = 0
+            for sx in subZ:
+                for sy in subZ:
+                    u =  c*(dx + sx) + s*(dy + sy)
+                    v = -s*(dx + sx) + c*(dy + sy)
+                    val += I0*math.exp(-0.5*((u/a)**2 + (v/b)**2))
+
+            if val < 0:
+                val = 0
+            gal.set(x, y, val/nsample**2)
+
+            I += val
+            Iuu += val*u**2
+            Ivv += val*v**2
+
+    Iuu /= I; Ivv /= I
+
+    exp = afwImage.makeExposure(afwImage.makeMaskedImage(gal))
+    exp.getMaskedImage().getVariance().setXY0(exp.getXY0()) # workaround #2577
+    exp.getMaskedImage().getVariance().set(1.0)
+    exp.setWcs(afwImage.makeWcs(afwCoord.Coord(0.0*afwGeom.degrees, 0.0*afwGeom.degrees),
+                                afwGeom.Point2D(0.0, 0.0), 1.0e-4, 0.0, 0.0, 1.0e-4))
+    psf = afwDetection.GaussianPsf(15, 15, 3.0)
+    exp.setPsf(psf)
+    return exp
+
+def makeSourceMeasurementConfig(forced=False, nsigma=6.0, nIterForRadius=1, kfac=2.5):
+    """Construct a SourceMeasurementConfig with the requested parameters"""
+    if forced:
+        msConfig = measBase.ForcedMeasurementConfig()
+        msConfig.algorithms.names = ["base_TransformedCentroid", "base_SdssShape", "extensions_photometryKron_KronFlux"]
+        msConfig.slots.centroid = "base_TransformedCentroid"
+    else:
+        msConfig = measBase.SingleFrameMeasurementConfig()
+        msConfig.algorithms.names = ["base_SdssCentroid", "base_SdssShape", "extensions_photometryKron_KronFlux"]
```

Comment:
Need to add `"base_SkyCoord"` here (NB not in line 97) or the test will fail when `display=True`; see discussion on JIRA.